### PR TITLE
Update email frow via email

### PR DESCRIPTION
**Pull request by smyzin**
- created a new hook that takes params from URL
- updated flow with taken params
- updated negative flow (account doesn't connected)

image
image

### CODE REVIEW

File: src/components/proofs/ProofList.tsx [hunk 1]
```diff
@@ -29,8 +31,18 @@ const bottomWrapper = classnames(
 )
 
 export function ProofListSuspended() {
-  const [category, setCategory] = useState<CategoriesTitles>('NFTs')
+  const params = useUrlParams()
+
+  const [category, setCategory] = useState<CategoriesTitles>(
```
Comment (MixailE) [inline src/components/proofs/ProofList.tsx:36]: Let's make a parameter where a specific section will be specified, without such checks

File: src/components/proofs/ProofList.tsx [hunk 2]
```diff
@@ -29,8 +31,18 @@ const bottomWrapper = classnames(
...
+    params ? 'Email' : 'NFTs'
+  )
   const { hasAnyProof } = useProofStore()
+  if (params) {
```
Comment (MixailE) [inline src/components/proofs/ProofList.tsx:40]: It seems to me that such logic should be in the rendering of sections, and not in the general file

Comment (smyzin) [reply src/components/proofs/ProofList.tsx:40]: Did it

File: src/components/proofs/ProofList.tsx [hunk 3]
```diff
@@ -28,8 +28,14 @@ const bottomWrapper = classnames(
...
 
-export function ProofListSuspended() {
-  const [category, setCategory] = useState<CategoriesTitles>('NFTs')
+export function ProofListSuspended({
+  selectedCategory,
+}: {
+  selectedCategory?: CategoriesTitles
+}) {
+  const [category, setCategory] = useState<CategoriesTitles>(
+    selectedCategory || 'NFTs'
```
Comment (MixailE) [inline src/components/proofs/ProofList.tsx:37]: can we specify the first category from the object here?

File: src/components/proofs/ConnectAccount.tsx [hunk 4]
```diff
@@ -49,6 +51,13 @@ const hintWrapper = classnames(
 
 export default function () {
   const { walletLoading, needNetworkChange } = useSnapshot(WalletStore)
+  const params = useUrlParams()
+  if (params) {
```
Comment (T-Damer) [inline src/components/proofs/ConnectAccount.tsx:55]: Let's remove `{}` from here

File: src/components/proofs/EmailProof.tsx [hunk 5]
```diff
@@ -81,12 +81,29 @@ const questionBlock = (open: boolean) =>
...
+
+  useEffect(() => {
+    if (defaultToken && defaultDomain) {
+      EmailDomainStore.emailDomain = defaultDomain
+      setToken(defaultToken ?? '')
+      setError(undefined)
+      setDomain(defaultDomain)
+    }
+  }, [defaultDomain, defaultToken])
+  console.log(EmailDomainStore.emailDomain, defaultDomain)
```
Comment (T-Damer) [inline src/components/proofs/EmailProof.tsx:106]: I think we don't need this `log`

File: src/components/proofs/EmailProof.tsx [hunk 6]
```diff
@@ -144,6 +162,7 @@ export default function () {
         {open && (
           <EmailProofForm
             domain={domain}
+            token={EmailDomainStore.emailDomain === defaultDomain ? token : ''}
```
Comment (T-Damer) [inline src/components/proofs/EmailProof.tsx:165]: Why don't we use `emailDomain` here? Without accessing `EmailDomainStore` I mean

File: src/components/proofs/EmailProof.tsx [hunk 7]
```diff
@@ -81,12 +81,29 @@ const questionBlock = (open: boolean) =>
...
+  const { emailDomain } = useSnapshot(EmailDomainStore)
+  const [domain, setDomain] = useState(defaultDomain ?? '')
+  const [open, setOpen] = useState(!!defaultDomain)
   const [error, setError] = useState<string | undefined>()
   const [generationStarted, setGenerationStarted] = useState(false)
-  const { emailDomain } = useSnapshot(EmailDomainStore)
+  const [token, setToken] = useState(defaultToken ?? '')
+
+  useEffect(() => {
+    if (defaultToken && defaultDomain) {
```
Comment (T-Damer) [inline src/components/proofs/EmailProof.tsx:99]: `if (!defaultToken || !defaultDomain) return`
Less nesting = better code, usually, let's keep it

File: src/components/proofs/EmailProof.tsx (hunk 5 above)
Comment (smyzin) [reply src/components/proofs/EmailProof.tsx:106]: Right, thx

File: src/components/proofs/EmailProof.tsx [hunk 8]
```diff
@@ -81,12 +81,29 @@ const questionBlock = (open: boolean) =>
...
-  const [open, setOpen] = useState(false)
+export default function ({
+  domain: defaultDomain,
+  token: defaultToken,
+}: {
+  domain?: string
+  token?: string
+}) {
+  const { emailDomain } = useSnapshot(EmailDomainStore)
+  const [domain, setDomain] = useState(defaultDomain ?? '')
```
Comment (T-Damer) [inline src/components/proofs/EmailProof.tsx:92]: So, `defaultDomain` may be empty? Btw, you're doing this operation twice, worth extracting into `const`

File: src/components/proofs/EmailProof.tsx (hunk 6 above)
Comment (smyzin) [reply src/components/proofs/EmailProof.tsx:165]: Changed it, thx

File: src/components/proofs/EmailProof.tsx (hunk 7 above)
Comment (smyzin) [reply src/components/proofs/EmailProof.tsx:99]: Changed it too

File: src/components/proofs/ProofList.tsx [hunk 9]
```diff
@@ -28,8 +28,12 @@ const bottomWrapper = classnames(
   alignItems('items-end')
 )
 
-export function ProofListSuspended() {
-  const [category, setCategory] = useState<CategoriesTitles>('NFTs')
+export function ProofListSuspended({
+  selectedCategory,
+}: {
+  selectedCategory: CategoriesTitles
```
Comment (T-Damer) [inline src/components/proofs/ProofList.tsx:34]: I think it's better to extract this one into `interface`

Comment (smyzin) [reply src/components/proofs/ProofList.tsx]: Done

File: src/components/proofs/ProofList.tsx [hunk 10]
```diff
@@ -64,10 +68,14 @@ export function ProofListSuspended() {
   )
 }
 
-export default function () {
+export default function ({
+  selectedCategory,
+}: {
+  selectedCategory: CategoriesTitles
```
[the remark this answers is not in the document]
Comment (smyzin) [reply src/components/proofs/ProofList.tsx]: Done

File: src/components/proofs/EmailProof.tsx [hunk 11]
```diff
@@ -81,12 +81,28 @@ const questionBlock = (open: boolean) =>
...
```
Comment (T-Damer) [inline src/components/proofs/EmailProof.tsx:92]: You're checking `??` in `EmailProofSection` - the only place where we use `<EmailProof />`, I think that you need to:
1) Remove `?` from arguments
2) Remove checks from here, you're passing 100% string

**Review by T-Damer — APPROVED** [review]

**Pull request merged**